### ISSUE TITLE
check_rel_metres & filter_inconsistent fail in get_galah if relevant columns are not in the data_map

### ISSUE THREAD

**Issue by Calamanthus** [Acanthiza/envImport#10]
Using a data_map in get_galah without the relevant columns for check_rel_metres (coordinateUncertaintyInMetres & generalisationInMetres) or filter_inconsistent (occurrenceStatus & organismQuantity) fails, because these columns have not been downloaded and don't exist. 

Perhaps we just need: `if(check_rel_metres & c("coordinateUncertaintyInMetres", "generalisationInMetres") %in% names(temp))`?

Otherwise, we'd have to download these relevant cols  even if they're not specified in the data_map to make sure check_rel_metres and filter_inconsistent can be performed, and then remove them later...

For now, I will workaround this by setting check_rel_metres and filter_inconsistent to FALSE (although it's not ideal, as the data will not be consistent with other downloads with the full data_map that have these initial 'cleaning' steps performed).

**Comment by Calamanthus** [comment]
Also, FYI, filter_inconsistent is still relevant with galah data, and using the ALA and CSDM data quality profiles does not fix the inconsistency, e.g. : `galah::galah_call() %>%
 galah::galah_apply_profile(ALA) %>%
 galah::galah_apply_profile(CSDM) %>%
 dplyr::filter((occurrenceStatus == "PRESENT" &
                   !is.na(organismQuantity) &
                   organismQuantity == 0
  )
  ) %>% 
  galah::atlas_counts()`
A tibble: 1 × 1
  count
  <int>
1  5677

**Comment by Acanthiza** [comment]
Try now @Calamanthus . I've just pushed some fixes to deal with NULL data_map and navigating the if statements when that is the case.

**Comment by Calamanthus** [comment]
Those changes should fix the issue, thanks. Although, as intimated above, I now have inconsistent status data in the download with no way of identifying and filtering that. I tried adding those filters to the qry to restrict the download, but it fails if you also include the other filter arguments I'm using for some reason. But, that is a separate issue, which I'll tinker with. Closing this issue now.
